### PR TITLE
fix(status): freeze elapsed, mark partial output, classify gate refusals

### LINKED ISSUE #29: Status and tool output fixes
- [ ] errorSummary reads up to 32 MiB to produce a 2 KB tail (internal/manager/status.go:110-119 via readFile). Worse: for err files over 32 MiB the io.LimitReader keeps the FIRST 32 MiB, so the reported "tail" is from the wrong end of the stream. Stat + read the last 2 KB directly.
- [ ] Status re-reads the entire out file on every poll of a terminal job (status.go:66,83), including agy_cancel's follow-up state lookup (tools.go:141) which only needs the state.
- [ ] readFile collapses IO errors into "" (status.go:97-108): an unreadable out file makes a completed job report "interrupted", or done with an empty result.
- [ ] ExitSpawnFail (127) renders as "exit 127:" with an empty stderr tail (status.go:90-92); the one failure mode that means "agy did not start" gets the least helpful message. Give it a dedicated case like the timeout has.
- [ ] Jobs recovered without a sentinel report plain done (status.go:65-71) with no marker that the output may be partial.
- [ ] Elapsed is time.Since(StartedAt) even for jobs that finished hours ago (status.go:46); no completion time is ever persisted. Persist an end timestamp.
- [ ] Gate rejection always blames "a conflicting agy job for this conversation or directory" even when the global cap was the cause (manager.go:164-166); tryAcquire returns a bare bool. Distinguish the two causes in the error.
- [ ] agy_cancel can return state "unknown" (tools.go:140), which no tool description documents.
- [ ] conversation_id and continue_latest can both be supplied; continue_latest silently overwrites the explicit id when the cache resolves, and precedence flips when it does not (tools.go:27-40, manager.go:157-161). Reject the combination or give the explicit id precedence.
- [ ] list_models discards agy's stderr from the error (internal/manager/models.go:11-15), so failures surface as a bare "exit status 1"; include (*exec.ExitError).Stderr. parseModels also treats any stdout line as a model name; fine for agy 1.0.7's bare list, worth pinning with a test.

### PR DESCRIPTION
**Pull request by tphakala**
Finishes the remaining items of #29. The other items (the 32 MiB stderr-tail bug, IO-error handling, the spawn-fail message, the documented agy_cancel "unknown" state, and list_models stderr) were already resolved in #42; this PR covers the rest.

## What changed

**`agy_cancel` no longer reads the full out file just to get a state (item 2)**
Added `Manager.State`, a state-only accessor. Every actual cancel leaves a non-zero sentinel (SIGTERM/SIGINT, timeout, spawn-fail), so the common path decides the state from the exit code alone with no out/err read. A clean exit (code 0), whose done-vs-failed split depends on whether the out file is readable, is deferred to `Status` so `State` can never disagree with it (including `Status`'s "successful but unreadable output -> failed" downgrade).

Note: this targets the `agy_cancel` lookup the issue called out. `agy_status` still reads the result on each poll of a terminal job by design, since the caller wants the result; in practice clients stop polling once a job is terminal.

**Recovered-without-sentinel output is marked partial (item 5)**
A job whose process died before writing a completion sentinel but left an out file was reported plain `done`. It now sets `Partial`, surfaced on the `agy_status` wire output (`"partial": true`) and noted in the tool description, so a caller does not treat possibly-truncated output as a complete result.

**Elapsed is frozen at completion (item 6)**
`Elapsed` was `time.Since(StartedAt)` even for jobs that finished hours ago. It is now measured to the `exit_code` sentinel's mtime (`jobstore.CompletedAt`), falling back to `time.Since` when the completion time is missing or implausibly precedes `StartedAt` (clock skew). This derives the end time from the write-once sentinel rather than adding a new persisted field or a supervisor write.

**Gate refusals name the real cause (item 7)**
`gate.tryAcquire` returned a bare bool, so a refusal always blamed "a conflicting agy job for this conversation or directory" even when the global concurrency cap was the cause. It now returns an `acquireOutcome` (`acquireOK` / `acquireKeyBusy` / `acquireAtCap`) and `StartJob` emits a distinct message per cause. The per-key check runs first, so when both apply the more specific key conflict wins (telling the user to wait for capacity would be misleading there).

**`conversation_id` + `continue_latest` are rejected together (item 9)**
Supplying both was ambiguous: `continue_latest` would silently overwrite the explicit id, and the precedence even flipped depending on whether the cache resolved. `StartJob` now rejects the combination up front (before the platform check, so the contract is enforced on every OS).

## Tests
- `TestGateRejectionDistinguishesCause` + updated gate tests for the new outcome type.
- `TestStatusElapsedFrozenAtCompletion` (pins the sentinel mtime and asserts ~frozen elapsed, not `time.Since`).
- `TestStatusInterruptedAfterReboot` now asserts `Partial`; `TestStatusDone` asserts a clean job is not partial.
- `TestStateMatchesStatusState` (State agrees with Status across every exit code) and `TestStateMatchesStatusOnUnreadableCleanExit` (the clean-exit-but-unreadable-output edge case, surfaced during peer review).
- `TestStartJobRejectsConversationIDWithContinueLatest`.

`go vet`, `golangci-lint` (0 issues), and `go test -race ./...` all pass.

Closes #29

## Summary by CodeRabbit

* **New Features**
  * Job completion timestamp estimation from on-disk sentinels.
  * Lightweight state accessor for faster job state checks.
  * MCP tools expose a "partial" flag when recovered without a completion sentinel.

* **Bug Fixes**
  * Concurrency responses now distinguish same-key conflicts from global capacity limits.
  * Rejections for invalid parameter combinations are returned immediately.
  * Job elapsed time freezes at completion.

* **Tests**
  * Expanded tests for concurrency causes, capacity, and partial/recovered statuses.

### CODE REVIEW

File: internal/jobstore/store.go [hunk 1]
```diff
@@ -237,6 +237,22 @@ func (s *Store) WriteExitCode(id string, code int) error {
...
+func (s *Store) CompletedAt(id string) (time.Time, bool) {
+	if !validJobID(id) {
+		return time.Time{}, false
+	}
+	info, err := os.Stat(filepath.Join(s.jobDir(id), "exit_code"))
+	if err != nil {
+		return time.Time{}, false
+	}
+	return info.ModTime(), true
+}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/jobstore/store.go:258]: Done in 6ab5fdc. CompletedAt now falls back to the out then err file mtime when the exit_code sentinel is absent, so a recovered job has an end timestamp.

File: internal/manager/status.go [hunk 2]
```diff
@@ -81,6 +86,7 @@ func (m *Manager) Status(id string) (Status, error) {
 	case out != "":
 		st.State = StateDone
 		st.Result = out
+		st.Partial = true // recovered without a sentinel; the output may be truncated
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/manager/status.go:93]: Done in 6ab5fdc. The no-sentinel terminal block now freezes elapsed via frozenElapsed before classifying the outcome, so a recovered job's elapsed stops growing.

File: internal/manager/status.go [hunk 3]
```diff
@@ -125,6 +134,63 @@ func (m *Manager) statusFromExitCode(dir string, meta jobstore.Meta, st Status,
...
+func (m *Manager) frozenElapsed(meta jobstore.Meta, running time.Duration) time.Duration {
+	end, ok := m.store.CompletedAt(meta.ID)
+	if !ok {
+		return running
+	}
+	if d := end.Sub(meta.StartedAt); d >= 0 {
+		return d
+	}
+	return running
+}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/manager/status.go:198]: Done in 6ab5fdc. frozenElapsed now clamps to 0 under clock skew instead of returning the ever-growing running duration, since the job is terminal.

**Comment by tphakala** [comment]
@coderabbitai Addressed the gate-outcome nitpick in 6ab5fdc: `StartJob` now captures the outcome and has a `default` case that fails closed with an "unknown acquire outcome" error, so a future `acquireOutcome` cannot fall through and start a job without a reserved slot or key.
